docs/development/tracing.md

```diff
@@ -0,0 +1,46 @@
+# Tracing
+
+SensorBucket processes large amounts of valuable data. To ensure no data is lost all data points are traced within the system.
+
+### Step
+
+Every pipeline message that appears on the message queue is evaluated by the Tracing Service and stored as a 'step' in the database. A step has the following format:
+
+| Column         | Description                                                                                    |
+| -------------- | ---------------------------------------------------------------------------------------------- |
+| TracingID      | ID used to match a data point to the step                                                      |
+| StepIndex      | Index of the current step to be executed from the pipeline steps array in the pipeline message |
+| StepsRemaining | Counts the remaining steps left for the data point                                             |
+| StartTime      | Time at which the pipeline message appears on the Message Queue                                |
+| Error          | If an error has occurred in the step it is stored here                                         |
+
+### Configuration
+The tracing service requires the following environment variables to be set in order to run:
+
+| Variable                    | Description                                              |
+| --------------------------- | -------------------------------------------------------- |
+| DB_DSN                      | The connection string for the PostgreSQL database        |
+| AMQP_HOST                   | The RabbitMQ host                                        |
+| AMQP_QUEUE_PIPELINEMESSAGES | The queue on which pipeline messages appear              |
+| AMQP_QUEUE_ERRORS           | The queue on which any errors produced by workers appear |
```

Comment:
Perhaps add optional variables for automatically binding the queue's to an exchange with topic. See the ingress archiver

Reply:
solved in https://github.com/sensorbucket/SensorBucket/pull/30/